src/ContentClient.ts

```diff
@@ -49,12 +51,19 @@ export class ContentClient implements ContentAPI {
     deploymentBuilderClass?: typeof DeploymentBuilder
   ) {
     this.contentUrl = sanitizeUrl(contentUrl)
+    this.deploymentBuilderClass = deploymentBuilderClass ?? DeploymentBuilder
     this.fetcher =
       fetcher ??
       new Fetcher({
         headers: getHeadersWithUserAgent('content-client')
       })
-    this.deploymentBuilderClass = deploymentBuilderClass ?? DeploymentBuilder
+
+    if (PROOF_OF_WORK) {
+      setImmediate(async () => {
```

Comment:
- I'd like to understand the reasoning behind `setImmediate` vs simply `await`
- `setImmediate` does not handle exceptions. If the async function fails it would cascade to the event loop and close in future versions and trigger a `unhandledRejection` in the process
- `setImmediate` is not available in all JS engines, i.e. it is not present in firefox

Reply:
The rationale behind using `setImmediate` was to avoid blocking the request with this logic, but after discussing this with @jmoguilevsky too I'm removing it and simply using an `await`.